src/Rule/IpHandler.php

```diff
@@ -80,116 +86,72 @@ private function getIpParsePattern(): string
-                    'attribute' => $context->getAttribute(),
-                    'value' => $value,
-                ],
-            );
-            return $result;
+            return self::getGenericErrorResult($rule->getMessage(), $context, $value);
         }
-        return $result;
+
+        return null;
     }
 
-    private function validateVersion(
+    private static function validateVersion(
         Ip $rule,
-        Result $result,
         int $ipVersion,
         string $value,
         ValidationContext $context
-    ): Result {
+    ): Result|null {
         if ($ipVersion === IpHelper::IPV6 && !$rule->isAllowIpv6()) {
-            $result->addError(
-                $rule->getIpv6NotAllowedMessage(),
-                [
-                    'attribute' => $context->getAttribute(),
-                    'value' => $value,
-                ],
-            );
-            return $result;
+            return self::getGenericErrorResult($rule->getIpv6NotAllowedMessage(), $context, $value);
         }
+
         if ($ipVersion === IpHelper::IPV4 && !$rule->isAllowIpv4()) {
-            $result->addError(
-                $rule->getIpv4NotAllowedMessage(),
-                [
-                    'attribute' => $context->getAttribute(),
-                    'value' => $value,
-                ],
-            );
-            return $result;
+            return self::getGenericErrorResult($rule->getIpv4NotAllowedMessage(), $context, $value);
         }
-        return $result;
+
+        return null;
     }
 
-    private function validateCidr(
+    private static function validateCidr(
         Ip $rule,
-        Result $result,
         ?string $cidr,
         string $ipCidr,
         string $value,
         ValidationContext $context
-    ): Result {
+    ): Result|null {
         if ($cidr !== null) {
             try {
                 IpHelper::getCidrBits($ipCidr);
             } catch (InvalidArgumentException) {
-                $result->addError(
-                    $rule->getWrongCidrMessage(),
-                    [
-                        'attribute' => $context->getAttribute(),
-                        'value' => $value,
-                    ],
-                );
-                return $result;
+                return self::getGenericErrorResult($rule->getWrongCidrMessage(), $context, $value);
             }
         }
+
         if (!$rule->isAllowed($ipCidr)) {
-            $result->addError(
-                $rule->getNotInRangeMessage(),
-                [
-                    'attribute' => $context->getAttribute(),
-                    'value' => $value,
-                ],
-            );
-            return $result;
+            return self::getGenericErrorResult($rule->getNotInRangeMessage(), $context, $value);
         }
-        return $result;
+
+        return null;
+    }
+
+    private static function getGenericErrorResult(string $message, ValidationContext $context, string $value): Result
```

Comment:
Why all private methods is static?

Reply:
Because these methods do not require current instance (`$this`).